### ISSUE TITLE
Cannot install gem because of HTTP 403 Forbidden

### ISSUE THREAD

**Issue by paulvt** [ntkme/sass-embedded-host-ruby#176]
When I try to use a project that tries to install the gem via bundle, installation fails:
```sh
$ bundle install
Fetching gem metadata from https://rubygems.org/............
Resolving dependencies...
Fetching sass-embedded 1.69.5 (x86_64-linux)

Retrying download gem from https://rubygems.org/ due to error (2/4): Gem::RemoteFetcher::FetchError bad response Forbidden 403 (https://rubygems.org/gems/sass-embedded-1.69.5-x86_64-linux.gem)

Retrying download gem from https://rubygems.org/ due to error (3/4): Gem::RemoteFetcher::FetchError bad response Forbidden 403 (https://rubygems.org/gems/sass-embedded-1.69.5-x86_64-linux.gem)

Retrying download gem from https://rubygems.org/ due to error (4/4): Gem::RemoteFetcher::FetchError bad response Forbidden 403 (https://rubygems.org/gems/sass-embedded-1.69.5-x86_64-linux.gem)
```

This is probably not an issue with this project itself, but I hope the issue can be raised somewhere somehow. Or maybe a re-upload helps?

**Comment by ntkme** [comment]
Debian unstable?

**Comment by ntkme** [comment]
What is the output of the following command:

``` sh
ruby -e 'puts Gem.platforms'
```

**Comment by paulvt** [comment]
> Debian unstable?

Yes!

```console
$ ruby -e 'puts Gem.platforms'
ruby
x86_64-linux

$ ruby -e 'puts Gem::Platform.new("x86_64-linux-gnu")'
x86_64-linux
```

Ah, so it should actually get the `...x86_64-linux-gnu.gem`?

**Comment by ntkme** [comment]
@paulvt See https://github.com/jekyll/jekyll/issues/9478#issuecomment-1785797746

**Comment by ntkme** [comment]
> Ah, so it should actually get the ...x86_64-linux-gnu.gem?

Yes. Debian unstable's rubygems 3.4.20 is patched by debian maintainers to match 3.3.15's behavior and it is broken. There is nothing I can do. You should report this via https://www.debian.org/Bugs/Reporting.

**Comment by paulvt** [comment]
I see. This seems to be done to not break all existing built/packaged native extensions with rubygems 3.3 (that stripped the `-gnu` suffix). This is going to be fixed when Ruby 3.2 enters Debian soon anyway. Thanks for the info!

**Comment by felipecsl** [comment]
Still cannot install sass-embedded, failing also on my github action:
```
Run ruby/setup-ruby@v1
Modifying PATH
Downloading Ruby
Extracting  Ruby
Print Ruby version
Installing Bundler
  Using Bundler 2.5.5 from Gemfile.lock BUNDLED WITH 2.5.5
  /opt/hostedtoolcache/Ruby/3.2.2/x64/bin/gem install bundler -v 2.5.5
  Successfully installed bundler-2.5.5
  1 gem installed
  Took   0.66 seconds
> bundle install
/opt/hostedtoolcache/Ruby/3.2.2/x64/bin/bundle config --local path /home/runner/work/lima.gl/lima.gl/vendor/bundle
/opt/hostedtoolcache/Ruby/3.2.2/x64/bin/bundle config --local deployment true
Cache key: setup-ruby-bundler-cache-v6-ubuntu-22.04-x64-ruby-3.2.2-wd-/home/runner/work/lima.gl/lima.gl-with--without--only--Gemfile.lock-330d2878ed272db[9](https://github.com/felipecsl/lima.gl/actions/runs/7584678335/job/20658884832?pr=883#step:6:10)dc133f2b439ef3b[23](https://github.com/felipecsl/lima.gl/actions/runs/7584678335/job/20658884832?pr=883#step:6:28)4387[28](https://github.com/felipecsl/lima.gl/actions/runs/7584678335/job/20658884832?pr=883#step:6:34)e0aafeb105e9dac61dd5c1477
/opt/hostedtoolcache/Ruby/3.2.2/x64/bin/bundle install --jobs 4
Fetching gem metadata from https://rubygems.org/.........
Your bundle is locked to sass-embedded (1.70.0) from rubygems repository
https://rubygems.org/ or installed locally, but that version can no longer be
found in that source. That means the author of sass-embedded (1.70.0) has
removed it. You'll need to update your bundle to a version other than
sass-embedded (1.70.0) that hasn't been removed in order to install.
Error: The process '/opt/hostedtoolcache/Ruby/3.2.2/x64/bin/bundle' failed with exit code 7
```

**Comment by ntkme** [comment]
@felipecsl You issue looks like a different one. It seems that you have the 1.70.0 platform ruby locked, which is clearly available on rubygems: https://rubygems.org/gems/sass-embedded/versions/1.70.0

I cannot reproduce it, so it might be a temporary glitch on one of rubygems' CDN mirror.

**Comment by suppadeliux** [comment]
I have the same 403 forbidden error in my fedora container with `ruby 3.1.2p20 (2022-04-12 revision 4491bb740a) [x86_64-linux]` and the latest update of this gem `1.70.0`

`
17:55:49 Fetching pathutil 0.16.2
17:55:49 Installing pathutil 0.16.2
17:55:49 Fetching sass-embedded 1.70.0 (x86_64-linux) (was 1.59.3)
17:55:49 
17:55:49 Retrying download gem from https://rubygems.org/ due to error (2/4): Gem::RemoteFetcher::FetchError bad response Forbidden 403 (https://rubygems.org/gems/sass-embedded-1.70.0-x86_64-linux.gem)
17:55:49 
17:55:49 Retrying download gem from https://rubygems.org/ due to error (3/4): Gem::RemoteFetcher::FetchError bad response Forbidden 403 (https://rubygems.org/gems/sass-embedded-1.70.0-x86_64-linux.gem)
17:55:49 Fetching kramdown 2.4.0
17:55:49 
17:55:49 Retrying download gem from https://rubygems.org/ due to error (4/4): Gem::RemoteFetcher::FetchError bad response Forbidden 403 (https://rubygems.org/gems/sass-embedded-1.70.0-x86_64-linux.gem)
17:55:49 Installing kramdown 2.4.0
`

**Comment by ntkme** [comment]
@suppadeliux It should never install `-linux` as there is only `-linux-gnu` and `-linux-musl`. If your rubygems is attempting to install `-linux` that means you have either a broken rubygems/bundler, or a Gemfile.lock previously created by a broken rubygems/bundler. Please try upgrade rubygems, remove Gemfile.lock and then try again.

**Comment by shawnyeager** [comment]
I'm running into the same problem with `sass-embedded`. I'm able to build my project locally on my M2 Mac, but I see the 403 errors when attempting to build on Netlify. This only started after running `bundle update` earlier this week. I assume there's been a gem update over the last week.

Delete my `Gemfile.lock`, running `gem update --system` and regenerating my .lock file does not resolve the problem. Still seeing the failures on Netlify.

**Comment by suppadeliux** [comment]
@shawnyeager  Since ruby installs also the latest version of  rubygems, have you tried to instead, update ruby, to the latest version?

**Comment by ntkme** [comment]
> I'm running into the same problem with sass-embedded. I'm able to build my project locally on my M2 Mac, but I see the 403 errors when attempting to build on Netlify in an Ubuntu 20 LTS container. This only started after running bundle update earlier this week. I assume there's been a gem update over the last week.

Likely means that the netlify Ubuntu container has an old version of rubygems. I'm not sure if you can upgrade that. Alternatively if you can upgrade to use a ruby 3.2 or 3.3 container, the default rubygems version from ruby `>=3.2` should always be good.

**Comment by shawnyeager** [comment]
@suppadeliux Changing `.ruby-version` to `3.2.3` fixed the build failures on Netlify. Thanks. 🙏🏼

**Comment by lsilvam** [comment]
where did you edit this `.ruby-version` to `3.2.3`?

**Comment by lsilvam** [comment]
> where did you edit this `.ruby-version` to `3.2.3`?

found it. also works for 3.3.0; with this version was able to deploy on netlify
